### ISSUE TITLE
the way of fine-tuning 

### ISSUE THREAD

**Issue by sunjia2015** [chainer/chainer#724]
I am a beginner of chainer. I have use my database to train a network from scratch, which get the same accuracy as caffe.
I want to use another database to train this network with the trained weights and  I changed the number of nodes of the fully connected layers. I write my network as follows and  model_base is used to fine-tuning.
Could any one tell me wheter my way is correct?
Thank you very much.

import chainer
import chainer.functions as F
import six
import six.moves.cPickle as pickle

class Age_gender(chainer.FunctionSet):

```
"""Single-GPU AlexNet without partition toward the channel axis.

"""
insize = 200

def __init__(self):
    super(Age_gender, self).__init__(            
        fc6=F.Linear(13824, 1024),
        fc7=F.Linear(1024, 1024),
        fc8=F.Linear(1024, 5),
    )

def forward(self, x_data, y_data, model_base, train=True):
    x = chainer.Variable(x_data, volatile=not train)
    t = chainer.Variable(y_data, volatile=not train)        
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv1(x))), 3,stride=2)
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv2(h))), 3, stride=2)
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv3(h))), 3, stride=2)
    h.unchain_backward()
    h = F.dropout(F.relu(self.fc6(h)))
    h = F.dropout(F.relu(self.fc7(h)))        
    h = self.fc8(h)
    return F.softmax_cross_entropy(h, t), F.accuracy(h, t)
```

**Comment by suranus** [comment]
sunjia2015,你好，我目前正要用chainer做训练，在配置环境上遇到一些问题，可以交流下吗？谢谢

**Comment by sunjia2015** [comment]
你有什么问题呢。

**Comment by delta2323** [comment]
@sunjia2015 

Your implementation works well as long as your intention is to fix `model_base`'s parameters and update only `Age_gender`'s ones. 

One suggestion is to use `volatile` attribute of Variable in order not to propagate error to parameters of `model_base`:
```
def forward(self, x_data, y_data, model_base, train=True):
    x = chainer.Variable(x_data, volatile=not train)
    x.volatile = 'on'
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv1(x))), 3, stride=2)
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv2(h))), 3, stride=2)
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv3(h))), 3, stride=2)
    h.volatile = 'off'
    h = F.dropout(F.relu(self.fc6(h)))
    ....
```

Your code constructs and destroys the part of computational graphs, which is inefficient.
If you switch the `volatile` attribute of `Variable`, `forward` method does not construct it from the scratch.

By the way, we ask developers and users to ask general questions at [User Group](https://groups.google.com/forum/#!forum/chainer).
If you do not mind, I would like you to submit there next time. Thank you for your cooperation.

**Comment by sunjia2015** [comment]
delta2323,thank you for your advice.
In your code , the part of y_data was deleted, which I think it is a mistake.
Could you tell me the reason?
I will submit my quesiton to User Group next time.

**Comment by delta2323** [comment]
@sunjia2015 
Ah, sorry. I omit the line about `y_data` just by mistake.

**Comment by sunjia2015** [comment]
delta2323
Thank you for your reply.
I write following code  according to your advice. Is there any problem?

def forward(self, x_data, y_data, model_base, train=True):
    x = chainer.Variable(x_data, volatile=not train)
    x.volatile = 'on'
    t = chainer.Variable(y_data, volatile=not train) 
    y.volatile = 'on'
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv1(x))), 3, stride=2)
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv2(h))), 3, stride=2)
    h = F.max_pooling_2d(F.relu(F.local_response_normalization(model_base.conv3(h))), 3, stride=2)
    h.volatile = 'off'
    h = F.dropout(F.relu(self.fc6(h)))
    ....

**Comment by suranus** [comment]
@sunjia2015 您是在linux系统上配置的吗？我在windows系统上一直不能安装成功

**Comment by sunjia2015** [comment]
@suranus .我用的是linux

**Comment by Fhrozen** [comment]
I am wondering if it is possible implement an Learning Rate (LR) constant initializer by layer. It is quite complicate to implement 2 or more optimizers for each part of the network at finetuning stage, and set up the LR on each optimizer.

**Comment by beam2d** [comment]
Chainer now supports configuring different hyperparameters for each parameter via `UpdateRule`.
